Initiate a proposal for the perf-dash

vhe182 opened the pull request:
Hi @meteorfox, here's is a draft of the proposal for the project we have been talking about.  This thing is in progress but I would like some input whenever you get the chance.  Thanks!

Parts worked on:
- Abstract
- Background
- Architecture Diagram

vhe182 committed:
Write Abstract and Background section proposal

Also included the call for the diagram.

vhe182 committed:
Create diagram for architecture of proposal

blinkmonster182 committed:
Fix broken link to diagram 1

Forgot about relative path...

meteorfox commented on docs/proposals/mvp-dashboard-design.md:
```diff
@@ -10,16 +10,37 @@ Last updated: 2017-01-12
+We propose a dashboard/server that allows for the automated retrieval,
+archive, format, and display of performance tests run in the performance lab.
+This tool will allow our team to investigate performance issues and facilitate
+the sharing of lab results with other teams.
```
The system should actually retrieve, and display performance test results from any lab. At the end of the day is just another filter in the query. 

What do you mean by 'format'?

Instead of 'performance test runs', we should say 'performance test results'.

I don't think the tool well help us in investigating issues, but instead actually find them. Track the performance over time across builds and different environment and configurations, doing so we will be able to spot regressions. I agree with sharing the results, but it's not just for the performance lab.

vhe182 replied:
@meteorfox: Thanks for the comments.  Here's my feedback.
1. Made the change concerning the actions the system does: retrieve and display 
1. I mean "format" as, the manner of displaying the data is changed from a json return, to a graph.
1. runs -> results
1. Okay, I'll change the wording so that it is understood that this tool is for all labs. I will change the wording about 'investigation' to identifying regressions. So this would be a monitoring system.

meteorfox commented on docs/proposals/mvp-dashboard-design.md:
```diff
@@ -10,16 +10,37 @@ Last updated: 2017-01-12
+1. Retrieve report file
+1. Format data so that it can be entered into a database
+1. Insert data into database
+1. Query the database for timerange of tests
```
We don't even do this today. We just manually fetch the data, format it and create the visualizations manually using ggplot2 or Excel.

vhe182 replied:
Made changes to better represent the actions taken for reviewing performance test results.

meteorfox commented on docs/proposals/mvp-dashboard-design.md:
```diff
@@ -10,16 +10,37 @@ Last updated: 2017-01-12
-the proposed change.
+Viewing performance test results from the performance lab as it trends across
+time is tedious and repetitive. To view this data, the following actions must
+take place:
```
Let's make the sentence not specific to the performance lab, but more general. Think about the steps if you had to do a performance tests on a QE lab as well.

vhe182 replied:
Done.  Generalized the target labs.

meteorfox commented on docs/proposals/mvp-dashboard-design.md:
```diff
@@ -10,16 +10,37 @@ Last updated: 2017-01-12
+
+This set of actions occurs everytime there is a need to investigate lab
+performance. Automating these actions allows more time for developers to
+invest in development and investigation of performance issues.
```
We should emphasize that our current approach has a very long feedback cycle. By the time, we collect the results and find issues, it's too late to address them in the current release. So, having a system that automatically displays the results as they are gathered, will help us shortened that feedback cycle and catch more issues **early**

vhe182 committed:
proposal: Make first set of requested changes

1. Made the change concerning the actions the system does: retrieve
and display
1. I mean "format" as, the manner of displaying the data is changed
from a json return, to a graph.
1. runs -> results
1. Changed the wording so that it is understood that this
tool is for all labs. I will change the wording about 'investigation'
to identifying regressions. So this would be a monitoring system.

vhe182 committed:
proposal: Make changes as requested in review

1. Generalize "performance lab" to all QE labs
1. Update list of actions so that it better reflects actual steps

vhe182 committed:
proposal: Make changes as requested in review

1. Add line talking about the cost that taking manual actions has
on finding performance issues
1. Change the wording to mention shortening of the feedback cycle.

vhe182 committed:
Merge branch 'proposal' of github.com:vhe182/rpc-perf-dash into proposal

meteorfox reviewed (commented):
LGTM, will you be adding more to the other sections?

vhe182 commented:
Yes, I think this PR is large enough though.  Maybe merge this one and I can do another PR for the rest?
